### DISCUSSION TITLE
Show offer type (buy/sell) i Market/Trades

### DISCUSSION THREAD

**Discussion by Akira45-0** [bisq-network/bisq#5466] [Ideas]
Hello. When I choose the Market option from the menu and then follow to Trades, I see the list of the most recent trades. It looks like this:
![grafik](https://user-images.githubusercontent.com/63785272/117031524-f2338500-ad00-11eb-9c4c-494923b0ee28.png)
On this list it is not shown whether the taken offer was a Buy offer or a Sell offer.
You may say that every trade has a buyer and a seller, but the problem is that when you post a Sell offer the whole amount of the trade and the deposit are immediately locked, while when posting a Buy offer only the deposit value is locked. The liquidity on certain markets is varying, to say the least, so I think it would be best if the traders could see how often and what value Sell offers are taken, before they decide to lock their coins, because disabling an offer which no one wants to take means losing the fee.
So I am asking for this offer type information (if it was Buy or Sell offer) to be added to the Trades list.

**Comment by Akira45-0** [comment]
I can see that it was present in the previous versions of Bisq. I don't know why it was removed, but I think it would be useful to bring it back.
![grafik](https://user-images.githubusercontent.com/63785272/117433061-59348200-af2b-11eb-862a-6eea6716bc52.png)

**Comment by ghost** [comment]
TradeStatistics was re-architected October 2020, removing data which consisted a privacy leak.  https://github.com/bisq-network/bisq/issues/3893  https://github.com/bisq-network/bisq/pull/4611  Part of the re-architecting was to reduce the data size since it was causing performance issues.

I am not sure if adding data to TradeStatistics causes a hard fork, meaning TradeStatistics3 would have to become TradeStatistics4.  (@chimp1984 would know).  There was a prior request by @pazza83 to add "percentage deviation from market" into TradeStatistics that has not been done either, for the same reasons.  I asked a [question about it](https://github.com/bisq-network/bisq/issues/4641#issuecomment-717223569) which went unanswered.

IMHO it is costly to change TradeStatistics and there would have to be really good reasons to justify the time and expense, and the maintainers would have to approve such a request first.

**Reply by chimp1984** [reply]
> IMHO it is costly to change TradeStatistics and there would have to be really good reasons to justify the time and expense, and the maintainers would have to approve such a request first.

Agree.

**Reply by m52go** [reply]
Regarding percent deviation from market, I always thought `primaryMarketTradePrice` was meant to be market price at the time, so that one can assess premium paid by comparing `primaryMarketTradePrice` with `tradePrice`. 

But as far as I can tell, `primaryMarketTradePrice` matches the `tradePrice` for all trades. Is this an error?

**Reply by ghost** [reply]
`primaryMarketTradePrice` is not stored in TradeStatistics3, rather it is `tradePrice` formatted appropriately for the relevant fiat or altcoin market.

https://github.com/bisq-network/bisq/blob/993f9b4e29a206dfe7ee76dc1882fca2fbc5c87d/core/src/main/java/bisq/core/trade/statistics/TradeStatisticsForJson.java#L71-L72

**Comment by Akira45-0** [comment]
Thank you for answering. I don't know much about Bisq internals and what time and expense would be needed. but I can say that it's bad for liquidity and for trading in general for the reasons I wrote above. So maybe it would be profitable anyway.

**Comment by pazza83** [comment]
Thanks @Akira45-0 for the post.

I agree that offer type would be good to be included.

I think the following would all be great to be included in the trade statistics:

- Percentage deviation from market
- Market price
- Trade Price
- Offer type

The ability to sort by payment method as well as currency would also be great. Or even filter by both payment method and currency.
